include/pocl.h

```diff
@@ -141,11 +143,28 @@ typedef struct
   void *device_ptr;
   void *host_ptr;
   size_t origin[3];
+  size_t h_origin[3];
   size_t region[3];
-  size_t rowpitch;
-  size_t slicepitch;
+  size_t h_rowpitch;
+  size_t h_slicepitch;
+  size_t b_rowpitch;
+  size_t b_slicepitch;
+  cl_mem buffer;
+} _cl_command_r_image;
```

Comment:
It would have been nice to keep the naming consistent with already existing command (_cl_command_read/write)